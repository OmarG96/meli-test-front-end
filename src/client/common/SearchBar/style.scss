@import "../../utils/scss/variables.scss";

.search-bar {
  display: flex;
  flex: 1;
  max-width: 60%;
}

.search-input {
  flex: 1;
  height: 32px;
  border-top-left-radius: 4px;
  border-bottom-left-radius: 4px;
  border: none;
  font-size: 18px;
  padding-left: 12px;

  &:focus {
    outline: none;
  }
}

.search-button {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 40px;
  border-top-right-radius: 4px;
  border-bottom-right-radius: 4px;
  border: none;
  background-color: $gray-1;

  &:hover {
    cursor: pointer;
  }
}
