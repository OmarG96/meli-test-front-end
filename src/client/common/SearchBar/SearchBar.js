import React, { useCallback, useState } from "react";
import { func } from "prop-types";

import iconSearch from "../../assets/images/ic_Search.png";
import "./style.scss";

const SearchBar = ({ onSearch = () => {} }) => {
  const [item, setItem] = useState("");

  const handleSubmit = useCallback(
    (event) => {
      event.preventDefault();
      onSearch(item);
    },
    [onSearch, item]
  );

  const handleChange = useCallback((event) => {
    const value = event.target.value;
    setItem(value);
  }, []);

  return (
    <form className="search-bar" onSubmit={handleSubmit}>
      <input
        name="search-input"
        value={item}
        onChange={handleChange}
        className="search-input"
        placeholder="Nunca dejes de buscar"
      />
      <button type="submit" className="search-button">
        <img src={iconSearch} />
      </button>
    </form>
  );
};

SearchBar.propTypes = {
  onSearch: func,
};

export default SearchBar;
