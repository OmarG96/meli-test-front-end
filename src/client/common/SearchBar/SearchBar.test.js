import React from "react";
import SearchBar from ".";
import { fireEvent, render } from "@testing-library/react";

describe("GIVEN SearchBar", () => {
  describe("WHEN is rendered", () => {
    it("THEN match to snapshot", () => {
      const { asFragment } = render(<SearchBar />);
      expect(asFragment()).toMatchSnapshot();
    });

    it("THEN should onSearch will be called with right params", () => {
      const onSearch = jest.fn();
      const { getByRole } = render(<SearchBar onSearch={onSearch} />);
      const searchbox = getByRole("searchbox");

      fireEvent.change(searchbox, { target: { value: "example" } });
      fireEvent.submit(searchbox);

      expect(onSearch).toBeCalledWith("example");
    });
  });
});
