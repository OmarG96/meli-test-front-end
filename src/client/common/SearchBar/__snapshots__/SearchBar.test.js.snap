// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN SearchBar WHEN is rendered THEN match to snapshot 1`] = `
<DocumentFragment>
  <form
    class="search-bar"
  >
    <input
      class="search-input"
      name="search-input"
      placeholder="Nunca dejes de buscar"
      role="searchbox"
      value=""
    />
    <button
      class="search-button"
      type="submit"
    >
      <img
        src="test-file-stub"
      />
    </button>
  </form>
</DocumentFragment>
`;
